ecosystem/scenarios/utils.py

```
@@ -0,0 +1,260 @@
+        }
+
+        # Execute setup.py
+        with open(setup_py, "r") as f:
+            code = compile(f.read(), str(setup_py), "exec")
+            exec(code, namespace)
+
+        # Parse requirements to extract package names
+        install_requires = set()
+        for req in requirements.get("install_requires", []):
+            # Extract package name (before any version specifier)
+            pkg_name = req.split(">=")[0].split("==")[0].split("<")[0].split(">")[0].split("[")[0].strip()
```

Pylint (3.3.7)

[convention] Line too long (106/100)

(C0301)